tests/components/zha/test_light.py

```diff
@@ -63,6 +67,46 @@
 }
 
 
+@asynctest.mock.patch(
+    "zigpy.zcl.clusters.general.OnOff.read_attributes", new=MagicMock()
+)
+async def test_light_refresh(hass, zigpy_device_mock, zha_device_joined_restored):
+    """Test zha light platform refresh."""
+
+    # create zigpy devices
+    zigpy_device = zigpy_device_mock(LIGHT_ON_OFF)
+    zha_device = await zha_device_joined_restored(zigpy_device)
+    on_off_cluster = zigpy_device.endpoints[1].on_off
+    entity_id = await find_entity_id(DOMAIN, zha_device, hass)
+
+    # allow traffic to flow through the gateway and device
+    await async_enable_traffic(hass, [zha_device])
+    on_off_cluster.read_attributes.reset_mock()
+
+    # not enough time passed
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(minutes=20))
+    await hass.async_block_till_done()
+    assert on_off_cluster.read_attributes.call_count == 0
+    assert on_off_cluster.read_attributes.await_count == 0
+    assert hass.states.get(entity_id).state == STATE_OFF
+
+    # 1 interval - 1 call
+    on_off_cluster.read_attributes.return_value = [{"on_off": 1}, {}]
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(minutes=80))
+    await hass.async_block_till_done()
+    assert on_off_cluster.read_attributes.call_count == 1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5156575)